manifests/git_repo.pp

```diff
@@ -7,7 +7,7 @@
 
   validate_string($source)
 
-  $sources = {"${upstream_remote_name}" => "https://github.com/${source}.git"}
+  $sources = hash([$upstream_remote_name, "https://github.com/${source}.git"])
```

Comment:
It generated an error on puppet 4 otherwise:

```
Could not parse for environment *root*: Illegal attempt to assign via [index/key]. Not an assignable reference
```

But your suggestion looks better I think, updated.